src/main/java/com/nutomic/syncthingandroid/syncthing/EventProcessor.java

```diff
@@ -0,0 +1,146 @@
+package com.nutomic.syncthingandroid.syncthing;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Environment;
+import android.preference.PreferenceManager;
+import android.support.v4.content.LocalBroadcastManager;
+import android.util.Log;
+import android.widget.Toast;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import eu.chainfire.libsuperuser.Shell;
+
+/**
+ * Run by the syncthing service to convert syncthing events into local broadcasts.
+ * It uses SyncthingService.GetEvents to read the pending events and wait for new events.
+ */
+public class EventProcessor implements SyncthingService.OnWebGuiAvailableListener {
+
+    private static final String TAG = "EventProcessor";
+    private static final String PREF_LASTSYNCID = "last_sync_id";
+
+    public static final String EVENT_BASE_URL = "com.nutomic.syncthingandroid.event";
```

Comment:
Why and how should it be removed? Actions for intents have to follow the naming convention specified within http://developer.android.com/guide/components/intents-filters.html (section Action). I don't see an exception for local itents within the documentation.
Maybe it won't matter. But I'd prefer to stick to the rules.